src/libs/Network/SequentialQueue.ts

```diff
@@ -194,6 +194,9 @@ function unpause() {
     const numberOfPersistedRequests = PersistedRequests.getAll().length || 0;
     Log.info(`[SequentialQueue] Unpausing the queue and flushing ${numberOfPersistedRequests} requests`);
     isQueuePaused = false;
+    if (numberOfPersistedRequests === 0) {
+        flushOnyxUpdatesQueue();
+    }
```

Comment:
@danieldoglas @tgolen can I have your opinion on this change please. I think you may know more about the possible implications of changing this behaviour in the queue.

Reply:
hmm. I'm not 100% sure. 

Considering that GetMissingOnyxUpdates is not added to the sequential queue, this might call the flush method while GetMissingOnyxUpdates is being executed.

@chrispader what do you think?

Reply:
In case it is of interest... another way of fixing the problem we are trying to fix here would be to stop forwarding the `CreateReportAction` command here: https://github.com/Expensify/Auth/blob/de9f50789e666b56e0bee0e1658b8fa67a35c742/auth/lib/Report.cpp#L7797-L7805 and do it without command forwarding. The command forwarding is causing the `OpenReport` to not return the correct `lastUpdateID`.

Reply:
I think it's definitely not idea to make `OpenReport` slower... but it also definitely sounds like a backend bug and the frontend fix is only a bandaid over it. I especially am concerned that the sequential queue is being changed to try and work around this.

I don't know if this is even possible, or what it would do... but what would happen if we changed `OpenReport` to not return `lastUpdateID` at all? I think that would be better than returning an incorrect one.

Reply:
> Considering that GetMissingOnyxUpdates is not added to the sequential queue, this might call the flush method while GetMissingOnyxUpdates is being executed.
> @chrispader what do you think?

Calling `flushOnyxUpdatesQueue()` while the `SequentialQueue` is still paused will result in nothing except for a log.

https://github.com/Expensify/App/blob/45c7e48568c7b2e902f801c5e7925ea76e76e20f/src/libs/Network/SequentialQueue.ts#L51-L54

Therefore i don't think the deferred update logic in `OnyxUpdateManager` would be affected. Though, this means that the `QueuedOnyxUpdates` won't actually get flushed if there is a `GetMissingOnyxUpdates` check currently running

Reply:
> but it also definitely sounds like a backend bug and the frontend fix is only a bandaid over it

I also agree with @tgolen on this. I think this behavior should be fixed on the backend, rather than updating the logic of the SequentialQueue

Reply:
@tgolen @chrispader This problem happens in many other places (see [my comment](https://github.com/Expensify/App/issues/49931#issuecomment-2451207417)). If we come up with a BE fix as suggested [here](https://github.com/Expensify/App/pull/51712#discussion_r1826308794), we need to fix other APIs, not only the OpenReport API

Reply:
> Calling flushOnyxUpdatesQueue() while the SequentialQueue is still paused will result in nothing except for a log.

@chrispader The solution here is we will call `flushOnyxUpdatesQueue` after the queue is unpaused and there is no request in the queue.

Reply:
> @chrispader The solution here is we will call `flushOnyxUpdatesQueue` after the queue is unpaused and there is no request in the queue.

True, my mistake. This should be fine imo. What i meant essentially is that this doesn't interfere with the current deferred update logic in `OnyxUpdateManager`, as long as we don't manually unpause the `SequentialQueue`

Reply:
@tgolen @chrispader What is next step here?

Reply:
@aldo-expensify Do you have any idea what the practical timing measurements are for not forwarding that command on the backend? Does it significantly slow down OpenReport or not?

Reply:
bump @aldo-expensify

Reply:
> @aldo-expensify Do you have any idea what the practical timing measurements are for not forwarding that command on the backend? Does it significantly slow down OpenReport or not?

This optimization was introduced here https://github.com/Expensify/Auth/pull/8595/files for the command [CreateChatReport](https://github.com/Expensify/Expensify/issues/299672), so it seems like affecting `OpenReport` was not the intention.

From the conversations in the [issue](https://github.com/Expensify/Expensify/issues/299672), it looks like the slow part of creating a new chat is checking if the "participants know each other" in this query: https://github.com/Expensify/Auth/blob/01acb32728415564d4ef0321934433bde38c1462/auth/lib/Account.cpp#L4465-L4484

Removing the command forwarding may help with @DylanDylann 's case [here](https://github.com/Expensify/App/issues/49931#issuecomment-2451207417) since this code is reused by more than one command and it may be root cause of that issue too.

Maybe one thing we can try is: calculate the `Account::isAccountKnownToCurrentUser` in `prePeek` (it is the expensive part we want to avoid in `process`) and revert the command forwarding.

Reply:
> Maybe one thing we can try is: calculate the `Account::isAccountKnownToCurrentUser` in `prePeek` (it is the expensive part we want to avoid in `process`) and revert the command forwarding.

I'll start working on a backend PR and assess how feasible this is.

Reply:
Started to work on this here: https://github.com/Expensify/Auth/pull/13168